pkg/reconciler/contour/contour.go

```diff
@@ -90,7 +91,22 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) r
 		// Reconcile an endpoint probe child kingress to ensure the Contour
 		// gateways have the endpoints for our generation's services.
 
-		desiredChIng := resources.MakeEndpointProbeIngress(ctx, ing)
+		// Fetch the HTTP Proxy resources from the PRIOR generation to include
+		// in the Endpoint Probe.  The Endpoint probe is used to warm new Envoy
+		// "clusters" (Endpoints), but also to keep the prior HTTP Proxy's "clusters"
+		// in existence until the new generation has been rolled out as fully ready.
+		var previousState []*v1.HTTPProxy
+		if selector, err := labels.Parse(fmt.Sprintf("%s=%s,%s!=%d",
+			resources.ParentKey, ing.Name,
+			resources.GenerationKey, ing.Generation)); err != nil {
+			return err
+		} else if elts, err := r.contourLister.HTTPProxies(ing.Namespace).List(selector); err != nil {
+			return err
+		} else if len(elts) != 0 {
```

Comment:
you don't really need `elts`  (or previousState variable) . Just assign to it.
Since the loop will be a noop in the MakeEndpointProbeIngress

Reply:
done